virtwho/virt/virt.py

```diff
@@ -492,6 +497,34 @@ def _get_data(self):
             reports[source_key] = report
         return reports
 
+    def _get_data_initial(self):
+        """
+        Waits for each source in self.source_keys to have a value returned from the datastore. This
+        does not check to see if the data has been previously sent. This method will wait for a
+        maximum of the self.interval period of time, then return whatever it has gathered thus far.
+        If data for each source key is gathered before the interval has expired, this method will
+        return.
+        @return: dict
+        """
+        reports = {}
+        while not reports and not self.is_terminated():
+            source_keys_remaining = list(self.source_keys)
+            time_waited = 0
+            while len(source_keys_remaining) > 0 and time_waited < self.interval and not self.is_terminated():
+                for source_key in source_keys_remaining:
+                    try:
+                        report = self.source.get(source_key)
+                        reports[source_key] = report
+                        source_keys_remaining.remove(source_key)
+                    except KeyError:
```

Comment:
Are you sure this is necessary? Looking at it, I thought initially it was for duplicate `source_key`(s), but remove will only remove one instance... If it is necessary, please leave a comment as to why.